TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/Drivebase.java

```diff
@@ -0,0 +1,275 @@
+package org.firstinspires.ftc.teamcode.hardware;
+
+import static org.firstinspires.ftc.teamcode.utility.Config.ENCODER_PER_INCH;
+
+import androidx.annotation.Nullable;
+
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.DcMotorSimple;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+import com.qualcomm.robotcore.hardware.IMU;
+
+import org.firstinspires.ftc.robotcore.external.Supplier;
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+import org.firstinspires.ftc.teamcode.utility.Api;
+
+/**
+ * Class that contains the entire drivebase for the robot. Contains methods to move the robot both in auto and in teleop.
+ */
+public class Drivebase {
+    private final DcMotor frdrive;
+
+    private final DcMotor fldrive;
+    private final DcMotor brdrive;
+    private final DcMotor bldrive;
+    private final IMU imu;
+    private final Supplier<Boolean> opModeIsActive;
+
+
+    /**
+     * Uses `hardwareMap` to initialize the motors and imu.
+     *
+     * @param hardwareMap    The hardware map to use for initialization.
+     * @param opModeIsActive Supplier that returns false if the program needs to stop.
+     */
+    public @Api Drivebase(HardwareMap hardwareMap, Supplier<Boolean> opModeIsActive) {
+        frdrive = hardwareMap.get(DcMotor.class, "FRDrive");
+        fldrive = hardwareMap.get(DcMotor.class, "FLDrive");
+        brdrive = hardwareMap.get(DcMotor.class, "BRDrive");
+        bldrive = hardwareMap.get(DcMotor.class, "BLDrive");
+        imu = hardwareMap.get(IMU.class, "IMU");
+        this.opModeIsActive = opModeIsActive;
+
+        fldrive.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        frdrive.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        bldrive.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        brdrive.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+
+        fldrive.setDirection(DcMotorSimple.Direction.REVERSE);
+        frdrive.setDirection(DcMotorSimple.Direction.FORWARD);
+        bldrive.setDirection(DcMotorSimple.Direction.REVERSE);
+        brdrive.setDirection(DcMotorSimple.Direction.FORWARD);
+
+        setMotorModes(DcMotor.RunMode.RUN_USING_ENCODER);
+    }
+
```

Comment:
In addition to a setMotorModes command, you could include a setTargetPosition command and a setPower command.

```java
    private void setMotorModes(DcMotor.RunMode mode) {
        fldrive.setMode(mode);
        frdrive.setMode(mode);
        bldrive.setMode(mode);
        brdrive.setMode(mode);
    }
```

Reply:
Added some methods to do this